### PR TITLE
Take frozen state into account when finishing trajectories.

### PR DESCRIPTION
**Pull request by MichaelGrupp**
Until now, the error response of an /finish_trajectory request for a
frozen trajectory was 'Trajectory ... is not created yet.'.

This is a lie. The new response is more accurate because the trajectory
__is__ created, but it just can't be finished because it's frozen.

### CODE REVIEW

File: cartographer_ros/cartographer_ros/node.cc [hunk 1]
```diff
@@ -443,6 +443,16 @@ bool Node::ValidateTopicNames(
 cartographer_ros_msgs::StatusResponse Node::FinishTrajectoryUnderLock(
     const int trajectory_id) {
   cartographer_ros_msgs::StatusResponse status_response;
+
+  // First, check if we can actually finish the trajectory.
+  if (map_builder_bridge_.GetFrozenTrajectoryIds().count(trajectory_id)) {
+    const std::string error =
+        "Trajectory " + std::to_string(trajectory_id) + " is frozen.";
+    LOG(INFO) << error;
```
Comment (ojura) [inline cartographer_ros/cartographer_ros/node.cc]: Nit: LOG(INFO) << "Error string"?

Comment (MichaelGrupp) [reply cartographer_ros/cartographer_ros/node.cc]: Should be LOG(ERROR), changed.

Comment (ojura) [reply cartographer_ros/cartographer_ros/node.cc]: Ah, didn't catch that. I was referring to creating a string variable - you can just stream the error message directly, without creating a variable.

Comment (MichaelGrupp) [reply cartographer_ros/cartographer_ros/node.cc]: Oh yes, also true :+1:

Comment (MichaelGrupp) [reply cartographer_ros/cartographer_ros/node.cc]: However, it's also true that the error string gets both logged and passed to the response, so the variable has a right to stay ;)

Comment (ojura) [reply cartographer_ros/cartographer_ros/node.cc]: Sorry, I missed that :)

**Comment by cschuet** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user cschuet. Merge queue now has a length of 2.

**Pull request merged**